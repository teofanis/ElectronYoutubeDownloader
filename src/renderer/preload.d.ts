import { IpcRequest, PromisedIpcResponse } from 'interfaces';
import { Channels } from 'main/preload';

declare global {
  interface Window {
    electron: {
      store: {
        getState: <T>() => T;
        setState: <T>(
          recipe: (draft: T) => void,
          description?: string | undefined
        ) => T;
        subscribe: <T>(
          listener: (state: T, description?: string | undefined) => void
        ) => () => void;
      };
      ipc: {
        send: <T>(
          channel: string,
          request: IpcRequest
        ) => PromisedIpcResponse<T>;
      };

      ipcRenderer: {
        removeAllListeners(channel?: Channels | string | null): void;
        sendMessage(channel: Channels, args: unknown[]): void;
        on(
          channel: string,
          func: (...args: unknown[]) => void
        ): (() => void) | undefined;
        once(channel: string, func: (...args: unknown[]) => void): void;
      };
    };
  }
}

export {};
