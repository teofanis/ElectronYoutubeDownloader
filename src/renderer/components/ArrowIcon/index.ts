// eslint-disable-next-line import/prefer-default-export
export { default as ArrowIcon } from './ArrowIcon';
