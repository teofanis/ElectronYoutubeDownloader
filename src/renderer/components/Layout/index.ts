/* eslint-disable import/prefer-default-export */
export { default as Layout } from './Layout';
