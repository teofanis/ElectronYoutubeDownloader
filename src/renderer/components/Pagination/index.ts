// eslint-disable-next-line import/prefer-default-export
export { default as Pagination } from './Pagination';
