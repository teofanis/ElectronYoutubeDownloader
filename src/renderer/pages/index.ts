import { Home } from './Home';

// eslint-disable-next-line import/prefer-default-export
export { Home };
