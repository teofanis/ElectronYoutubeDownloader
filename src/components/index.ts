// eslint-disable-next-line import/prefer-default-export
import { Button } from 'components/Button';
import { CancelButton } from 'components/CancelButton';
import { DownloadableItem } from 'components/DownloadableItem';
import { DownloadableItemControls } from 'components/DownloadableItemControls';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'components/DownloadButton';
import { Downloader } from 'components/Downloader';
import { DownloadItemControl } from 'components/DownloadItemControl';
import { FileInput } from 'components/FileInput';
import { Heading } from 'components/Heading';
import { InputError } from 'components/InputError';
import { Layout } from 'components/Layout';
import { Logo } from 'components/Logo';
import { ProgressBar } from 'components/ProgressBar';
import { TextInput } from 'components/TextInput';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloadableItemTransition,
  // Automatic export of components
};
