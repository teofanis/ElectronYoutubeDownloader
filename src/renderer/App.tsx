import { Home } from 'pages';
import { MemoryRouter as Router, Route, Routes } from 'react-router-dom';
import './global.css';

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
      </Routes>
    </Router>
  );
}
