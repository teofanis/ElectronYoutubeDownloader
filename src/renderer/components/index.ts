// eslint-disable-next-line import/prefer-default-export
import { Button } from 'renderer/components/Button';
import { CancelButton } from 'renderer/components/CancelButton';
import { DownloadableItem } from 'renderer/components/DownloadableItem';
import { DownloadableItemControls } from 'renderer/components/DownloadableItemControls';
import { DownloadableItemTransition } from 'renderer/components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'renderer/components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'renderer/components/DownloadButton';
import { Downloader } from 'renderer/components/Downloader';
import { DownloaderControls } from 'renderer/components/DownloaderControls';
import { DownloadItemControl } from 'renderer/components/DownloadItemControl';
import { FileInput } from 'renderer/components/FileInput';
import { Heading } from 'renderer/components/Heading';
import { InputError } from 'renderer/components/InputError';
import { Layout } from 'renderer/components/Layout';
import { Logo } from 'renderer/components/Logo';
import { ProgressBar } from 'renderer/components/ProgressBar';
import { TextInput } from 'renderer/components/TextInput';
import { Pagination } from 'renderer/components/Pagination';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloaderControls,
  Pagination,
  // Automatic export of components
};
