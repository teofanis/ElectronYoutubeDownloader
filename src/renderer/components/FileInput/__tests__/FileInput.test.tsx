import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { FileInput } from 'renderer/components';

const containerClasses = `text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2`;

const buttonClasses = `bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold`;

describe('FileInput Test', () => {
  it('should render correctly', () => {
    const rendered = render(
      <FileInput label="Choose a file" onClick={jest.fn} />
    );
    expect(rendered).toMatchSnapshot();
    const { container } = rendered;
    expect(container.querySelector('div')).not.toBeNull();
    expect(container.querySelector('button')).not.toBeNull();
    expect(container.querySelector('input')).not.toBeNull();

    expect(container.querySelector('button')).toHaveClass(buttonClasses);

    expect(container.querySelector('div')).toHaveClass(containerClasses);

    expect(container.querySelector('input')).toHaveAttribute('type', 'hidden');
    expect(container.querySelector('input')).toHaveAttribute(
      'style',
      'display: none;'
    );
  });

  it('should render correctly with errored', () => {
    const rendered = render(
      <FileInput label="Choose a file" onClick={jest.fn} errored />
    );
    expect(rendered).toMatchSnapshot();
    const { container } = rendered;
    expect(container.querySelector('div')).not.toBeNull();
    expect(container.querySelector('button')).not.toBeNull();
    expect(container.querySelector('input')).not.toBeNull();

    expect(container.querySelector('button')).toHaveClass(buttonClasses);

    expect(container.querySelector('div')).toHaveClass(
      `${containerClasses} border-red-500 right-2 outline-red-600 shadow-outline-red-500`
    );

    expect(container.querySelector('input')).toHaveAttribute('type', 'hidden');
    expect(container.querySelector('input')).toHaveAttribute(
      'style',
      'display: none;'
    );
  });

  it('should render correctly with value', () => {
    const rendered = render(
      <FileInput label="Choose a file" onClick={jest.fn} value="test" />
    );
    expect(rendered).toMatchSnapshot();
    const { container } = rendered;
    expect(container.querySelector('div')).not.toBeNull();
    expect(container.querySelector('button')).not.toBeNull();
    expect(container.querySelector('input')).not.toBeNull();
    expect(container.querySelector('span')).not.toBeNull();

    expect(container.querySelector('button')).toHaveClass(buttonClasses);

    expect(container.querySelector('div')).toHaveClass(containerClasses);

    expect(container.querySelector('input')).toHaveAttribute('value', 'test');
    expect(container.querySelector('input')).toHaveAttribute('type', 'hidden');
    expect(container.querySelector('input')).toHaveAttribute(
      'style',
      'display: none;'
    );
  });

  it('should render correctly with className', () => {
    const rendered = render(
      <FileInput
        label="Choose a file"
        onClick={jest.fn}
        className="test-class"
      />
    );
    expect(rendered).toMatchSnapshot();
    const { container } = rendered;
    expect(container.querySelector('div')).not.toBeNull();
    expect(container.querySelector('button')).not.toBeNull();
    expect(container.querySelector('input')).not.toBeNull();

    expect(container.querySelector('button')).toHaveClass(buttonClasses);

    expect(container.querySelector('div')).toHaveClass(
      `${containerClasses} test-class`
    );

    expect(container.querySelector('input')).toHaveAttribute('type', 'hidden');
    expect(container.querySelector('input')).toHaveAttribute(
      'style',
      'display: none;'
    );
  });

  it('should handle click', () => {
    const onClick = jest.fn();
    const rendered = render(
      <FileInput label="Choose a file" onClick={onClick} />
    );
    expect(rendered).toMatchSnapshot();
    const { container } = rendered;
    expect(container.querySelector('div')).not.toBeNull();
    expect(container.querySelector('button')).not.toBeNull();
    expect(container.querySelector('input')).not.toBeNull();

    expect(container.querySelector('button')).toHaveClass(buttonClasses);

    expect(container.querySelector('div')).toHaveClass(containerClasses);

    expect(container.querySelector('input')).toHaveAttribute('type', 'hidden');
    expect(container.querySelector('input')).toHaveAttribute(
      'style',
      'display: none;'
    );

    container.querySelector('button')?.click();
    expect(onClick).toBeCalled();
  });
});
