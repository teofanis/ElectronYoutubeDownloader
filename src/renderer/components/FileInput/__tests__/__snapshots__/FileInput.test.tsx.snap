// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput Test should handle click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              undefined"
      >
        <button
          class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
          type="button"
        >
          Choose a file
        </button>
        <input
          style="display: none;"
          type="hidden"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              undefined"
    >
      <button
        class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
        type="button"
      >
        Choose a file
      </button>
      <input
        style="display: none;"
        type="hidden"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FileInput Test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              undefined"
      >
        <button
          class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
          type="button"
        >
          Choose a file
        </button>
        <input
          style="display: none;"
          type="hidden"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              undefined"
    >
      <button
        class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
        type="button"
      >
        Choose a file
      </button>
      <input
        style="display: none;"
        type="hidden"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FileInput Test should render correctly with className 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              test-class"
      >
        <button
          class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
          type="button"
        >
          Choose a file
        </button>
        <input
          style="display: none;"
          type="hidden"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              test-class"
    >
      <button
        class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
        type="button"
      >
        Choose a file
      </button>
      <input
        style="display: none;"
        type="hidden"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FileInput Test should render correctly with errored 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              border-red-500 right-2 outline-red-600 shadow-outline-red-500 
              undefined"
      >
        <button
          class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
          type="button"
        >
          Choose a file
        </button>
        <input
          style="display: none;"
          type="hidden"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              border-red-500 right-2 outline-red-600 shadow-outline-red-500 
              undefined"
    >
      <button
        class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
        type="button"
      >
        Choose a file
      </button>
      <input
        style="display: none;"
        type="hidden"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FileInput Test should render correctly with value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              undefined"
      >
        <button
          class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
          type="button"
        >
          Choose a file
        </button>
        <span
          class="pl-2"
        >
          test
        </span>
        <input
          style="display: none;"
          type="hidden"
          value="test"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-black flex-1
              placeholder-gray-800
              mt-2
              text-base transition duration-500 ease-in-out transform border-transparent rounded-lg
              focus:border-blueGray-500
              bg-gray-300
              focus:bg-gray-100
              focus:outline-none
              focus:shadow-outline
              focus:ring-2
            ring-gray-700
              ring-offset-current ring-offset-2
              
              undefined"
    >
      <button
        class="
          bg-gray-400
          hover:bg-gray-300
          py-2.5
          rounded-lg px-4 z-10
          text-base
        text-black
          border
          border-blueGray-500
          font-semibold
          "
        type="button"
      >
        Choose a file
      </button>
      <span
        class="pl-2"
      >
        test
      </span>
      <input
        style="display: none;"
        type="hidden"
        value="test"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
